react-to-webcomponent.js

```diff
@@ -43,7 +43,12 @@ export default function (ReactComponent, React, ReactDOM, options = {}) {
 	var WebComponent = function () {
 		var self = Reflect.construct(HTMLElement, arguments, this.constructor);
 		if (options.shadow) {
-			self.attachShadow({ mode: 'open' });
+			if(typeof options.shadow === 'string') {
```

Comment:
Since the `true === 'open'` functionality is a backwards compatibility thing that will be deprecated in the future, I'd like to structure these conditions a bit differently and add a warning:

```
if (typeof options.shadow === 'string') {
  self.attachShadow({ mode: options.shadow });
} else if (options.shadow) {
  console.warn('Specifying the "shadow" option as a boolean is deprecated and will be removed in a future version.');
  self.attachShadow({ mode: 'open' });
}
```